src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs

```diff
@@ -75,6 +80,23 @@ public override void Write(Utf8JsonWriter writer, Version value, JsonSerializerO
             writer.WriteStringValue(span.Slice(0, charsWritten));
 #else
             writer.WriteStringValue(value.ToString());
+#endif
+        }
+
+        internal override Version ReadAsPropertyNameCore(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            return ReadCore(ref reader);
```

Comment:
nit: missing assert for property name token type like other overrides?

Reply:
I believe the `ReadCore` method does asserts of its own so adding one here felt redundant.